WalletWasabi/WabiSabi/Crypto/CredentialIssuer.cs

```diff
@@ -58,7 +58,7 @@ public class CredentialIssuer
 			MaxAmount = maxAmount;
 			RangeProofWidth = (int)Math.Ceiling(Math.Log2(MaxAmount));
 			CredentialIssuerSecretKey = Guard.NotNull(nameof(credentialIssuerSecretKey), credentialIssuerSecretKey);
-			NumberOfCredentials = Guard.InRangeAndNotNull(nameof(numberOfCredentials), numberOfCredentials, 1, 100);
+			NumberOfCredentials = Guard.InRangeAndNotNull(nameof(numberOfCredentials), numberOfCredentials, 1, 100); // What's up with this?
```

Comment:
It is a sanity check in case we make a mistake somewhere. Basically the point is that the `numberOfCredentials` could configurable and it could be stored in a config file for example. So, if we make a mistake and set a number of credentials out of that range then we will get an OutOfRangeException. 

I think currently is pretty clear we are going to use always **two** credentials and then it is a protocol constant.

Reply:
making it constant would keep the code simplest, and we can always convert it to a config parameter later if we find a good reason to, so assuming `CredentialsPerRequest` constant is added I would suggest something like:

```suggestion
			NumberOfCredentials = Guard.Equal(nameof(numberOfCredentials), numberOfCredentials, 1, CredentialsPerRequest);
```

Reply:
This is 2. Isn't it equal to `MaxInputCountByAlice`.

Reply:
they are independent: `MaxInputCountByAlice`determines how many inputs (presumably only for reused addresses, and i'm having second thoughts about supporting that, it seems to complicate the code substantially compared to the benefits), whereas the number of credentials determines how many credentials are presented and requested for each request, controlling the degree of fan-out or fan-in that is possible.

Reply:
I don't get it. There is no separate parameter for `CredentialsPerRequest` only `WabiSabiConfig.MaxInputCountByAlice`. Shall I add another parameter to the config as `CredentialsPerRequest`?

Reply:
not a config parameter but a constant, its value should be 2 and it's an exact requirement not a maximum value

Reply:
If it is a constant and we are checking it for equality in the ctor, then it should not be among the ctor parameters but a constant.

Reply:
Created a PR for this https://github.com/zkSNACKs/WalletWasabi/pull/5488